js_modules/dagit/packages/core/src/search/useRepoSearch.tsx

```diff
@@ -127,15 +127,15 @@ const secondaryDataToSearchResults = (data?: SearchSecondaryQuery) => {
 };
 
 export const useRepoSearch = () => {
-  const {data: bootstrapData, loading: bootstrapLoading} = useQuery<SearchBootstrapQuery>(
-    SEARCH_BOOTSTRAP_QUERY,
-    {
-      fetchPolicy: 'cache-and-network',
-    },
-  );
+  const [
+    performBootstrapQuery,
+    {data: bootstrapData, loading: bootstrapLoading},
+  ] = useLazyQuery<SearchBootstrapQuery>(SEARCH_BOOTSTRAP_QUERY, {
+    fetchPolicy: 'cache-and-network',
+  });
 
   const [
-    performQuery,
+    performSecondaryQuery,
     {data: secondaryData, loading: secondaryLoading, called: secondaryQueryCalled},
   ] = useLazyQuery<SearchSecondaryQuery>(SEARCH_SECONDARY_QUERY, {
```

Comment:
so, do we need to distinguish between the query and the secondary query now?

I thought that the main query we performed immediately and the secondary query was on interaction?

Reply:
I'm not sure why the initial query was being performed immediately. Did we have a good reason for that? I feel like that was maybe just an oversight on my part.

In my head I guess I was thinking it bootstrapped upon initial interaction, then performed the secondary query once a search was actually performed (which is still the case). I think maybe the secondary query is expected to be a lot more expensive, so keeping them separate might still be sensible.